### PR TITLE
See #7110. Target QQ replica count policy

### PR DESCRIPTION
**Pull request by SimonUnge**
## Proposed Changes

Add policy and operator policy for target qq replica count.
See #7110 

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [ ] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

_Put an `x` in the boxes that apply.
You can also fill these out after creating the PR.
If you're unsure about any of them, don't hesitate to ask on the mailing list.
We're here to help!
This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

### CODE REVIEW

**Comment by michaelklishin** [comment]
This adds a policy key and a configuration options but they are not used. Is the plan to use these values in later PRs? We can only merge such PRs into `main` then.

**Comment by SimonUnge** [comment]
Right, meant to be used for automatic qq group growth one that is figured out.

Would it make sense to add a flag to the grow command too, all|even| and something like "target"?

**Comment by michaelklishin** [comment]
There is no consensus on our team at the moment about this set of features, so it will have to wait.

**Comment by SimonUnge** [comment]
I was thinking of adding a little logic to the target-qq-replica-count, that if it is present on queue creation, use the max value of target-qq-replica-count, and x-quorum-initial-group-size. That is, update `get_default_quorum_initial_group_size` to check if a policy is present.

**Comment by SimonUnge** [comment]
Updated PR with logic taking policy into account on QQ creation

File: deps/rabbit/src/rabbit_quorum_queue.erl [hunk 1]
```diff
@@ -1581,12 +1609,18 @@ quorum_ctag(Other) ->
...
+        {undefined, undefined} ->
             application:get_env(rabbit, quorum_cluster_size, 3);
-        {_Type, Val} ->
-            Val
+        {undefined, V} ->
+            V;
+        {{_Type, V}, undefined} ->
+            V;
+        {{_Type, ArgV}, PolV} ->
+            max(ArgV, PolV)
```
Comment (SimonUnge) [inline deps/rabbit/src/rabbit_quorum_queue.erl:1623]: @michaelklishin @illotum 

Question: Should Policy have precedence over x-quorum-initial-group-size?

Comment (michaelklishin) [reply deps/rabbit/src/rabbit_quorum_queue.erl:1623]: Optional arguments historically have taken precedence. I now recall that we did not expose `quorum-initial-group-size` via policies intentionally at first. @kjnilsson do you remember why?

**Comment by michaelklishin** [comment]
Team RabbitMQ decided that this cannot be done in a "one step at a time" manner and needs a single issue that will serve as a design document. I will put together something that would capture our current understanding of the goal of dynamic QQ replica management with the goal of maintaining an operator-set minimum.

**Pull request closed without merging**